### PR TITLE
[CHK-427] Add DDD-style `Transaction` aggregate

### PR DESCRIPTION
**Pull request by giovanniberti**
#### List of Changes

* Add DDD-style `Transaction` aggregate

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/pagopa/transactions/model/Transaction.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package it.pagopa.transactions.model;
```
Comment (infantesimone) [inline src/main/java/it/pagopa/transactions/model/Transaction.java:1]: what do you think if we put all the `aggregate` and `valueObjet` in a different package? like domain?

Comment (giovanniberti) [reply src/main/java/it/pagopa/transactions/model/Transaction.java:1]: I'm ok with that, would it be everything besides `SoapEnvelope`? If it boils down to that we could even rename `model` → `domain` and put `SoapEnvelope` in some other package (just saying, I don't feel like `SoapEnvelope` belongs to "model")?

Comment (infantesimone) [reply src/main/java/it/pagopa/transactions/model/Transaction.java:1]: correct!

**Comment by infantesimone** [comment]
solved the coverage and some sonar issues, for me it's ok 🚀

**Review by MarcoAGreco — DISMISSED** [review]

**Pull request merged**